docs/SETUP_TLS_CERTIFICATES.md

```diff
@@ -0,0 +1,232 @@
+1. Install [certbot](https://certbot.eff.org/), as directed by Let's encrypt instructions:  
+    On Ubuntu:
+    ```
+    sudo add-apt-repository ppa:certbot/certbot
+    sudo apt-get update
+    sudo apt-get install certbot
+    ``` 
+    
+    On CentOS:
+    ```
+    wget https://dl.eff.org/certbot-auto
+    chmod a+x certbot-auto
+    ``` (CentOS)
+1. Request a certificate from Let's Encrypt using Certbot:
+    ```
+    mkdir -p tmp/certbot/conf.d tmp/certbot/work.d tmp/certbot/log.d
+    sudo certbot certonly -d frodo.buthey.net --standalone --config-dir tmp/certbot/conf.d/ --work-dir tmp/certbot/work.d/ --logs-dir tmp/certbot/log.d/ 
+    d=frodo.buthey.net; p=$(sudo find . -wholename "*live/$d"); for f in $(sudo ls -1 $p |grep .pem); do sudo cp $p/$f $(basename $p)-$f; done; sudo chown $(whoami) $d-*.pem
+    
+    ```
+    Resulting files:
+    * `frodo.buthey.net-privkey.pem` - private key file for domain `frodo.`
+    * `frodo.buthey.net-cert.pem` - certificate file for domain `frodo.`
+    * `frodo.buthey.net-chain.pem` - a certificate chain representing Let's encrypt CA (a certificate root)
+    
+## Installing certificates in system (Ubuntu, CentOS)
+
+This guide shows how to install root CA certificates in system. Having root CA certificates installed in system reduces complexity of TLS setup.
+
+### Installing root certificate in Ubuntu
+
+1. Obtain a certificate root file (e.g. `snaptest-CA.crt` from the guides in this document)
+1. Install the certificate in the system:
+    ```
+    sudo cp snaptest-ca.crt /usr/local/share/ca-certificates/
+    sudo update-ca-certificates --verbose --fresh
+    ```
+    Sample output:
+    ```
+    Clearing symlinks in /etc/ssl/certs...
+    done.
+    Updating certificates in /etc/ssl/certs...
+    Doing .
+    WARNING: Skipping duplicate certificate ACCVRAIZ1.pem
+    WARNING: Skipping duplicate certificate ACCVRAIZ1.pem
+    173 added, 0 removed; done.
+    ```
+
+### Installing root certificate in CentOS
+
+1. Obtain a certificate root file (e.g. `snaptest-ca.crt` from the guides in this document)
+1. Install the certificate tooling and certificate in the system:
+    ```
+    yum install ca-certificates
+    update-ca-trust enable
+    cp /opt/snap/certs/snaptest-ca.crt /etc/pki/ca-trust/source/anchors/
+    update-ca-trust extract
+    ```
+
+## Testing TLS-secured plugin communication in local setup
+
+We'll assume the following files are all available:
+* `snaptest-ca.crt` - certificate root - installed in system
+* `snaptest-srv.key` - server's private key file
+* `snaptest-srv.crt` - server's certificate file
+* `snaptest-cli.key` - framework's private key file
+* `snaptest-cli.crt` - framework's certificate file
+
+### Starting a telemetry workflow:
+
+1. Start a new shell session and execute `snapteld` with certificate and key generated for client:
+    ```
+    snapteld -t 0 -l 1 --tls-cert=/tmp/snaptest-cli.crt --tls-key=/tmp/snaptest-cli.key
```

Comment:
> I don't see it's supported in snaptel now.

Do you mean: `snaptel plugin load --plugin-cert=/tmp/snaptest-srv.crt --plugin-key=/tmp/snaptest-srv.key snap-plugin-collector-rand` ?
It's already done.

The document I'm proposing here, `SETUP_TLS_CERTIFICATES.md` relates to the Snap as available on GH. The instructions are valid for API v1 which is still the official one.
I'll be happy to contribute documentation updates regarding plugin security once API v2 becomes publicly available.